### ISSUE TITLE
CI Testing

### ISSUE THREAD

**Issue by IamLunchbox** [meffie/molecule-proxmox#22]
Would you consider automating tox tests of changes to this repository through a CI pipeline?  

In the current setup that would deduct either CI minutes from your personal github account though. Maybe that changes for orgs, but I don't know more until i check the docs. 

Otherwise, this would require a public proxmox instance, which i could take a look at. And as a last resort I could either host a drone runner easily or take a look at the self hosting of other runners, e.g. github.

**Comment by meffie** [comment]
That would be helpful.  A real test would require a proxmox instance, so I'm not sure
how that could be setup.   

Currently, I run the pytests via tox against a proxmox instance running in my homelab. 

The environment variables required to run the tests are listed in the README file.

________________________________________
From: IamLunchbox ***@***.***>
Sent: Friday, September 20, 2024 11:30 AM
To: meffie/molecule-proxmox
Cc: Subscribed
Subject: [meffie/molecule-proxmox] CI Testing (Issue #22)

Would you consider automating tox tests of changes to this repository through a CI pipeline?

In the current setup that would deduct either CI minutes from your personal github account though. Maybe that changes for orgs, but I don't know more until i check the docs.

Otherwise, this would require a public proxmox instance, which i could take a look at. And as a last resort I could either host a drone runner easily or take a look at the self hosting of other runners, e.g. github.

**Comment by IamLunchbox** [comment]
Yeah, i do tests similarly. And a cloud instance with paravirtualisation would either be very slow or expensive - unless theres a ci hook, which spawns and configures the node on demand.

I checked the github runner docs. Especially since a github runner does not require inbound https traffic i am inclined to go that route and provide access to a limited pool on my node.

But first ill need to go through the security docs of self hosted runners. There are some pitfalls, especially with foreign PRs and their potentially malicious code.

Sep 20, 2024 21:19:18 Michael Meffie ***@***.***>:

**Comment by meffie** [comment]
For the CI pipeline, perhaps we could somehow use pytest to mock the proxmox api we use. It would not be an end-to-end test, but it would give us some code coverage.

**Comment by IamLunchbox** [comment]
Yeah maybe that is easier for now - or rather for the foreseeable future - i think this is not a high priority issue.

**Comment by meffie** [comment]
Currently using tox and a local proxmox for CI testing, so going to close this for now.
